### PR TITLE
feat<CI>:Integration of Gitlab CI for build and tests.

### PR DESCRIPTION
**Pull request by yudaykiran**
**What this PR does / why we need it**:
- Added the gitlab yaml to track commits and build in gitlab
- Gitlab yaml triggers subsequent pipelines to run litmus tests.

**Which issue this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*: fixes #

**Special notes for your reviewer**:
Signed-off-by: Uday Kiran Y <uday.kiran@openebs.io>

### CODE REVIEW

File: .gitlab-ci.yml [hunk 1]
```diff
@@ -0,0 +1,106 @@
...
+    - ./configure --enable-code-coverage=yes --enable-debug
+    - make --no-print-directory -s pkg-utils pkg-kmod
+    - sudo dpkg -i *.deb
+    - make cstyle
+    - sudo /sbin/modprobe zfs
+    - /sbin/ztest
+
+baseline-image:
+  stage: baseline
+  image: atulabhi/kops:v10
```
Comment (vishnuitta) [inline .gitlab-ci.yml:82]: can we change this to openebs dockerhub or quay.io?

**Comment by vishnuitta** [comment]
looks to be fine with respect to compilation and building image.
@atulabhi can you look at baseline-image section?

**Review by vishnuitta — APPROVED** [review]
compilation and image build process looks good.

File: .gitlab-ci.yml (hunk 1 above)
Comment (yudaykiran) [reply .gitlab-ci.yml:82]: @vishnuitta , we will going forward, a Dockerfile for a image called openebs/infra has been developed, which is present in the openebs/test-tools repo, but there are still a couple of issues in its push scripts, did not want to hold off the zfs builds for that.

**Review by atulabhi — APPROVED** [review]

**Pull request merged**